ui/app/src/state/epics/contentTypes.ts

```diff
@@ -65,21 +65,21 @@ export default [
     action$.pipe(
       ofType(fetchComponentsByContentType.type, setContentTypeFilter.type),
       withLatestFrom(state$),
-      switchMap(([, state]) =>
-        fetchItemsByContentType(
+      switchMap(([, state]) => {
+        const allowedContentTypes = Object.entries(state.preview.guest?.allowedContentTypes)
+          .filter(([, value]) => value.shared)
+          .map(([key]) => key);
+        return fetchItemsByContentType(
           state.sites.active,
-          state.preview.components.contentTypeFilter === 'all'
-            ? Object.values(state.contentTypes.byId)
-                .filter(
-                  (contentType: ContentType) =>
-                    contentType.type === 'component' && !contentType.id.includes('/level-descriptor')
-                )
-                .map((contentType) => contentType.id)
+          state.preview.components.contentTypeFilter === 'compatible'
+            ? allowedContentTypes?.length
+              ? allowedContentTypes
+              : [''] // Empty array is like sending no filters, so it returns items unfiltered.
```

Comment:
This is not an empty array so either the comment or the object needs to be fixed.